server/mvcc/backend/backend.go

```diff
@@ -124,6 +126,9 @@ type BackendConfig struct {
 	UnsafeNoFsync bool `json:"unsafe-no-fsync"`
 	// Mlock prevents backend database file to be swapped
 	Mlock bool
+
+	// Hooks are getting executed on each Backend transaction
+	Hooks Hooks
```

Comment:
`Hooks` doesn't seem really descriptive... Are we planning to extend this to other pre- or post-commit operations? How about `OnPreCommit` or `ForEach` (to be consistent with bbolt API)?

Reply:
I imagine adding pre-transaction (OnTransactionStart) hook as well, that would contain additional validation code 
(enabled particularly in tests), that e.g. verifies whether initial cindex (or revision) in DB matches the state cached in RAM. 

I would like to keep container (type) for this as it allows to cleanly organize the code/tests in a separate file (within the same package). 

Options: 
  - TxHooks (https://en.wikipedia.org/wiki/Hooking)
  - Triggers (https://en.wikipedia.org/wiki/Database_trigger)
  - OnEventHandlers 
  - Flatten the structure.

Reply:
@gyuho - pls let me know which of this options would make the purpose clear ?

Reply:
@gyuho - could you, please, take a look to let me finish the PR

Reply:
That makes sense. LGTM, thanks!